packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -0,0 +1,107 @@
+ *
+ * @remarks
+ *
+ * Note: to initialize the tree's data for initial creation, implementers of this class will need to override {@link PureDataObject.initializingFirstTime} and set the data in {@link TreeDataObject.tree}.
+ *
+ * @example Implementing `initializingFirstTime`
+ *
+ * ```typescript
+ * protected override async initializingFirstTime(): Promise<void> {
+ * 	this.tree.initialize(...);
+ * }
+ * ```
+ *
+ * @privateRemarks
+ * TODO: Before promoting this beyond internal, we should consider alternative API patterns that don't depend on
+ * sub-classing and don't leak Fluid concepts that should ideally be internal.
+ * See `tree-react-api` for an example of a pattern that avoids unnecessary leakage of implementation details.
+ *
+ * @internal
+ */
+export abstract class TreeDataObject<
+	TSchema extends ImplicitFieldSchema = ImplicitFieldSchema,
+> extends PureDataObject {
+	/**
+	 * The configuration used to initialize new documents, as well as to interpret (schematize) existing ones.
+	 *
+	 * @remarks
+	 * The fact that a single view schema is provided here (on the data object) makes it impossible to try and apply multiple different schema.
+	 * Since the view schema currently does not provide any adapters for handling differences between view and stored schema,
+	 * it's also impossible for this single view schema to handle multiple different stored schema.
+	 * Therefore, with this current API, two different applications (or different versions of the same application)
+	 * with differing stored schema requirements (as implied by their view schema) can not collaborate on the same tree.
+	 * The only schema evolution that's currently possible is upgrading the schema to one that supports a superset of what the old schema allowed,
+	 * and collaborating between clients which have view schema that exactly correspond to that stored schema.
+	 * Future work on tree as well as these utilities should address this limitation.
+	 */
+	public abstract readonly config: TreeViewConfiguration<TSchema>;
+
+	/**
+	 * View of the underlying tree.
+	 */
+	#treeView: TreeView<TSchema> | undefined;
+
+	/**
+	 * Gets the view of the underlying {@link @fluidframework/tree#ITree}.
+	 *
+	 * @throws
+	 * If the tree has not yet been initialized, this will throw an error.
+	 */
+	public get tree(): TreeView<TSchema> {
+		if (this.#treeView === undefined) {
+			throw new UsageError(uninitializedErrorString);
+		}
+		return this.#treeView;
+	}
+
+	public override async initializeInternal(existing: boolean): Promise<void> {
+		if (existing) {
+			// data store has a root tree so we just need to set it before calling initializingFromExisting
+			const channel = await this.runtime.getChannel(treeChannelId);
+			if (!SharedTree.is(channel)) {
+				throw new Error(
+					`Content with id ${channel.id} is not a SharedTree and cannot be loaded with treeDataObject.`,
+				);
+			}
+			const sharedTree = channel as ITree;
```

Comment:
This explicit cast should ne unneeded.
```suggestion
			const sharedTree: ITree = channel;
```